### PR TITLE
enhance: Pre-allocate ids for compaction

### PR DESCRIPTION
**Pull request by bigsheeper**
This PR removes the dependency of compaction on the ID allocator by pre-allocating the logID and segmentID.

issue: https://github.com/milvus-io/milvus/issues/33957

### CODE REVIEW

**Comment by bigsheeper** [comment]
rerun ut

**Comment by bigsheeper** [comment]
/hold 
also need to pre-allocate segmentID for mix compaction

**Comment by bigsheeper** [comment]
rerun ut

**Comment by bigsheeper** [comment]
rerun ut

File: internal/datanode/services.go [hunk 1]
```diff
@@ -226,22 +230,25 @@ func (node *DataNode) CompactionV2(ctx context.Context, req *datapb.CompactionPl
 		task = compaction.NewLevelZeroCompactionTask(
 			taskCtx,
 			binlogIO,
-			node.allocator,
 			node.chunkManager,
 			req,
 		)
 	case datapb.CompactionType_MixCompaction:
+		if req.GetPreAllocatedSegments() == nil || req.GetPreAllocatedSegments().GetBegin() == 0 {
```
Comment (czs007) [inline internal/datanode/services.go:237]: len(req.GetPreAllocatedSegments()) == 0 ?

Comment (bigsheeper) [reply internal/datanode/services.go:237]: PreAllocatedSegments is not a slice

**Comment by bigsheeper** [comment]
rerun ut

**Pull request merged**